searx/results.py

```diff
@@ -156,43 +156,48 @@ def __init__(self):
         self.corrections = set()
         self._number_of_results = []
         self.engine_data = defaultdict(dict)
-        self._ordered = False
+        self._closed = False
         self.paging = False
         self.unresponsive_engines = set()
         self.timings = []
         self.redirect_url = None
+        self.on_result = on_result or (lambda _: True)
 
     def extend(self, engine_name, results):
+        if self._closed:
+            return
+
         standard_result_count = 0
         error_msgs = set()
         for result in list(results):
             result['engine'] = engine_name
-            if 'suggestion' in result:
+            if 'suggestion' in result and self.on_result(result):
                 self.suggestions.add(result['suggestion'])
-            elif 'answer' in result:
+            elif 'answer' in result and self.on_result(result):
```

Comment:
Another point I do not understand, why `self.on_result(result)` is called, I would expect a call with the arguments needed by the plugin's `on_result(request, search, result).  May be I mix up things do not understand the concept in deep / sorry.

Reply:
https://github.com/dalf/searxng/blob/2b96ace61118d0a94eb74f6d078728a0c73276de/searx/search/__init__.py#L179-L193

* `ResultContainer.on_result = SearchWithPlugins_on_result(self, result)`
* `SearchWithPlugins_on_result(self, result)` calls `plugins.call(self.ordered_plugin_list, 'on_result', self.request, self, result)`

The idea is to keep everything about the plugins in one place: the SearchWithPlugins class.